### PR TITLE
 fix(connectInfiniteHits): fix #2928 

### LINKED ISSUE #2928: duplicated inifinite hits with slow network
**Describe the bug 🐛**
When using the infinite hits widget on a slow network, clicking `show more` button may load duplicated hits.

**To Reproduce 🔍**
1. Go to https://community.algolia.com/instantsearch.js/v2/dev-novel/?selectedStory=InfiniteHits.default
2. Force your network settings to a slow network (example 3G)
3. Click on "Show more"
4. Scroll down to the hits
5. See the duplicated entries

**Expected behavior 💭**
Do not load/display duplicated hits.

**Screenshots 🖥**
image

**Environment:**
Slow network

### PR DESCRIPTION
**Pull request by bobylito**
When there is a slow network, the stalled search mechanics will trigger a new render with the same content. This has generally no impact. However the infinite builds up a list of all the results, and there was no mechanics in place to check if the same page has been already received.

To test, you can use dev-novel and with the performance set to slow 3g, you can see that there are no duplicates.

Fix #2928 

Might also be a patch for #2938 (need confirmation)

### CODE REVIEW

File: src/connectors/infinite-hits/connectInfiniteHits.js [hunk 1]
```diff
@@ -115,7 +118,10 @@ export default function connectInfiniteHits(renderFn, unmountFn) {
           results.hits = escapeHits(results.hits);
         }
 
-        hitsCache = [...hitsCache, ...results.hits];
+        if (lastReceivedPage < state.page) {
```
Comment (Haroenv) [inline src/connectors/infinite-hits/connectInfiniteHits.js]: we'll need to be careful with this line if we ever allow "scrolling up" as in #1972

Comment (bobylito) [reply src/connectors/infinite-hits/connectInfiniteHits.js]: Do you think it's blocking for now?

File: src/connectors/infinite-hits/__tests__/connectInfiniteHits-test.js [hunk 2]
```diff
@@ -185,8 +196,81 @@ describe('connectInfiniteHits', () => {
...
+      state: helper.state,
+      helper,
+      createURL: () => '#',
+    });
+
+    expect(rendering).toHaveBeenCalledTimes(3);
+    expect(rendering).toHaveBeenLastCalledWith(
+      expect.objectContaining({
+        hits: [{ objectID: 'a' }, { objectID: 'b' }],
+        results: expect.any(Object),
```
[the remark this answers is not in the document]
Comment (bobylito) [reply src/connectors/infinite-hits/__tests__/connectInfiniteHits-test.js]: I wasn't sure what was the good strategy, that's why it's a bit inconsistent 😅

File: src/connectors/infinite-hits/connectInfiniteHits.js (hunk 1 above)
Comment (Haroenv) [reply src/connectors/infinite-hits/connectInfiniteHits.js]: I think for now it's fine, we just need to keep it in mind if we ever allow scrolling to go the other direction

File: src/connectors/infinite-hits/__tests__/connectInfiniteHits-test.js [hunk 3]
```diff
@@ -54,22 +59,28 @@ describe('connectInfiniteHits', () => {
       createURL: () => '#',
     });
 
-    // test that rendering has been called during init with isFirstRendering = false
-    expect(rendering.callCount).toBe(2);
```
[the remark this answers is not in the document]
Comment (bobylito) [reply src/connectors/infinite-hits/__tests__/connectInfiniteHits-test.js]: No. Good catch.

**Comment by bobylito** [comment]
Did the changes @samouss :)

File: src/connectors/infinite-hits/connectInfiniteHits.js (hunk 1 above)
Comment (bobylito) [reply src/connectors/infinite-hits/connectInfiniteHits.js]: Ok thanks :)

**Pull request merged**